src/main/kotlin/com/nftco/flow/sdk/cadence/json-cadence.kt

```diff
@@ -322,26 +415,64 @@ open class ArrayField(value: Array<Field<*>>) : Field<Array<Field<*>>>(TYPE_ARRA
 
 open class DictionaryField(value: Array<DictionaryFieldEntry>) : Field<Array<DictionaryFieldEntry>>(TYPE_DICTIONARY, value) {
     constructor(value: Iterable<DictionaryFieldEntry>) : this(value.toList().toTypedArray())
+
     companion object {
         fun fromPairs(value: Iterable<Pair<Field<*>, Field<*>>>): DictionaryField {
             return DictionaryField(value.map { DictionaryFieldEntry(it) }.toTypedArray())
         }
+
         fun <K, V> fromMap(value: Map<K, V>, keys: (K) -> Field<*>, values: (V) -> Field<*>): DictionaryField {
             return fromPairs(value.mapKeys { keys(it.key) }.mapValues { values(it.value) }.map { Pair(it.key, it.value) })
         }
     }
 }
+
 open class DictionaryFieldEntry(val key: Field<*>, val value: Field<*>) : Serializable {
     constructor(pair: Pair<Field<*>, Field<*>>) : this(pair.first, pair.second)
 }
 
-open class AddressField(value: String) : Field<String>(TYPE_ADDRESS, if (!value.lowercase().startsWith("0x")) { "0x$value" } else { value }) {
+open class AddressField(value: String) : Field<String>(
+    TYPE_ADDRESS,
+    if (!value.lowercase().startsWith("0x")) {
+        "0x$value"
+    } else {
+        value
+    }
+) {
     constructor(bytes: ByteArray) : this(bytes.bytesToHex())
 }
 
 @kotlinx.serialization.Serializable
-open class PathValue(val domain: String, val identifier: String) : Serializable
-open class PathField(value: PathValue) : Field<PathValue>(TYPE_PATH, value)
+open class PathValue(val domain: String, val identifier: String) : Serializable {
+    override fun hashCode(): Int {
```

Comment:
Updated hashCode and equality check for PathField and PathValue to fix test cases